ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py

```diff
@@ -181,6 +182,11 @@ def test_avg_pool2d(
     fn_name,
 ):
     dtype, x, kernel, stride, pad = x_k_s_p
+
+    data_format = random.choice(["NCHW", "NHWC"])
```

Comment:
you shouldn't use random to generate any of the arguments, that might give you Flaky errors. all the arguments should be generated through hypothesis.